src/js/player.js

```diff
@@ -274,8 +275,10 @@ class Player extends Component {
     // Add a style element in the player that we'll use to set the width/height
     // of the player in a way that's still overrideable by CSS, just like the
     // video element
-    this.styleEl_ = document.createElement('style');
-    el.appendChild(this.styleEl_);
+    this.styleEl_ = stylesheet.getStyleElement('vjs-styles-dimensions');
+    let defaultsStyleEl = document.querySelector('.vjs-styles-defaults');
+    let head = document.querySelector('head');
+    head.insertBefore(this.styleEl_, defaultsStyleEl ? defaultsStyleEl.nextSibling : head.firstChild);
```

Comment:
I don't think it's worth worrying about if there's no head. I think looking for the `defaultsStyleEl` and prepending to HEAD otherwise, is quick and easy enough to do.